File: algorithms/sorting/bubbleSort.cpp
#include <vector>
#include <iostream>
// Time: O(N^2)
// Space: O(1)
class BubbleSort
{
public:
    std::vector<int> sort(std::vector<int> nums)
    {
        for (int i = 0; i < nums.size(); i++)
        {
            for (int j = 0; j < nums.size() - 1; j++)
            {
                if (nums[j] > nums[j + 1])
                {
                    int temp = nums[j];
                    nums[j] = nums[j + 1];
                    nums[j + 1] = temp;
                }
            }
        }

        return nums;
    }
};

int main()
{
    std::vector<int> list = {5, 4, 7, 9, 3, 8, 1, 0, 2};

    BubbleSort s;

    std::vector<int> sortedList = s.sort(list);

    for (int num : sortedList)
    {
        std::cout << num << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: algorithms/sorting/bubbleSort.cs
using System;

public static void BubbleSort(int[] nums)
{
    for (int i = 0; i < nums.Length; i++)
    {
        for (int j = 0; j < nums.Length - 1; j++)
        {
            if (nums[j] > nums[j + 1])
            {
                int temp = nums[j];
                nums[j] = nums[j + 1];
                nums[j + 1] = temp;
            }
        }
    }
}

public static void Main()
{
    int[] numbers = { 42, 7, 19, 88, 4, 73, 25, 11, 39, 2 };

    Console.WriteLine("Unordered Array:");
    foreach (int number in numbers)
    {
        Console.Write(number + " ");
    }
    Console.WriteLine();

    BubbleSort(numbers);

    Console.WriteLine("Ordered Array:");
    foreach (int number in numbers)
    {
        Console.Write(number + " ");
    }
    Console.WriteLine();
}

